server/server/Controllers/adminCrud.js

```diff
@@ -101,7 +101,109 @@ const removeOneOrManyItems = async (req, res) => {
   };
 
 
+//  Admin add new Restaurant
+const addnewRestaurant = async (req, res) => {
+  try {
+    const { name, location, phoneNumber, restaurantPicture, acceptedPayment, category } =
+      req.body;
+
+    const newRestaurant = new restaurantsModel({
+      name,
+      location,
+      phoneNumber,
+      restaurantPicture,
+      acceptedPayment,
+      category,
+    });
 
+    await newRestaurant.save();
+
+    res
+      .status(201)
+      .json({ message: "Restaurant added successfully", newRestaurant });
+  } catch (error) {
+    console.error("Error adding restaurant:", error);
+    res.status(500).json({ message: error.message });
+  }
+};
+
+
+// Admin get restaurant by id
+const getRestaurantById = async (req, res) => {
+const { id } = req.params;
+try {
+    const restaurant = await restaurantsModel.findById(id);
+    if (!restaurant) {
+        // If the restaurant is not found
+        return res.status(404).json({ message: "Restaurant not found" });
+    }
+    // If the restaurant is found
+    res.json(restaurant);
+} catch (err) {
+    // If there's an error
+    res.status(500).json({ message: err.message });
+}
+};
+
+// admin update restaurant details
+const updateRestaurant = async (req, res) => {
+try {
+   const { id } = req.params; 
+   const { name, location, phoneNumber, restaurantPicture, acceptedPayment, category } =
+   req.body;
+
+ const restaurant  = new restaurantsModel.findById(id);
+ if (!restaurant) {
+  return res.status(404).json({ message: "Restaurant not found" });
+}
+// Update details
+restaurant.name = name;
+restaurant.location = location;
+restaurant.phoneNumber = phoneNumber;
+restaurant.restaurantPicture = restaurantPicture;
+restaurant.acceptedPayment = acceptedPayment;
+restaurant.category = category;
+
+ 
+const updatedRestaurant = await restaurant.save();
```

Comment:
It should be just like the item update controller 


```suggestion
 const restaurant  = await restaurantsModel.findByIdAndUpdate(id,{ list updated properties here },{ runValidators: true, new: true });
 if (!restaurant) {
  return res.status(404).json({ message: "Restaurant not found" });
}
```